### PR TITLE
Request URI handling refactoring / Multi-license support / JSON post refactoring / Multi-copyright support

### PR DESCRIPTION
**Pull request by robfrawley**
Notable changes:
- Request URI ordering no longer important; the following are equal:
  - `/2010-2030/license.txt/69a991e`
  - `/69a991e/@2016/license.txt`
  - `/@2010-2016/69a991e/license.txt`
- Date range can have "@" proceeding it (previously only singular date could have "@" proceeding):
  - `/@2010-2016` == `/2010-2016`
  - `/@2010` == `/2010`
- New date format added that displays date in URI through current year:
  - `/@2010-` == `2010 - 2016` in template output
- License can now be specified (only MIT and ISC are have template ATM) using format `+NAME` in URI:
  - `/+ISC` gives you `licenses/ISC.html`
  - `/+MIT` gives you `licenses/MIT.html`
- The above can also be specified in JSON using `license` key:
  - `{ "license": "MIT" }` or `{ "license": "ISC" }`
- JSON HTTP POST's now returns valid HTTP response codes and JSON
  - Post updates Git
- Support for more than one copyright holder, per remy/mit-license#764

Reference: remy/mit-license#765 / remy/mit-license#73 / remy/mit-license#764

Examples:

GET **`http://rmf.mit-license.org/+ISC/@2010-/license.txt`**:

OUTPUTS:

```
ISC License (ISC)
Copyright © 2010-2016 Rob M Frawley 2nd, http://robfrawley.com <source@robfrawley.com>
[...]
```

---

GET **`http://rmf.mit-license.org/@2010-2012/license.txt`**:

OUTPUTS:

```
The MIT License (MIT)
Copyright © 2010-2012 Rob M Frawley 2nd, http://robfrawley.com <source@robfrawley.com>
[...]
```

---

GET **`http://rmf.mit-license.org/@2010-2012`**:

WITH **`rmf.json`**:

``` json
{
  "copyright" : [
    {
      "name" :     "Rob M Frawley 2nd",
      "url" :      "http:\/\/robfrawley.com",
      "email" :    "source@robfrawley.com",
      "gravatar" : true
    },
    {
      "name" :  "First Last",
      "email" : "flast@domain.com"
    }
  ],
  "format" : "txt"
}
```

OUTPUTS:

```
The MIT License (MIT)
Copyright © 2010-2012 Rob M Frawley 2nd <source@robfrawley.com>, First Last <flast@domain.com>
[...]
```

### CODE REVIEW

File: licenses/ISC.html [hunk 1]
```diff
@@ -0,0 +1,69 @@
+<!DOCTYPE html>
+<html id="home" lang="en">
+<head>
+<title>MIT License</title>
```
Comment (b264) [inline licenses/ISC.html]: ISC License

Comment (robfrawley) [inline licenses/ISC.html]: Good catch; fixed.

**Comment by robfrawley** [comment]
Is there any interest in merging this @remy ? I'll re-base it if there is...

**Comment by remy** [comment]
I am interested, but I've not really digested the url changes (or whether they're changes).

Also, the git integration (the POSTing to JSON) needs a bit of a re-think since it's now all hosted on Heroku (which means the disk data is ephemeral/totally lost on each push), so github PRs is the only route at the moment.

**Comment by robfrawley** [comment]
Regarding the URLs, the new handling is 100% BC (as far as I know). It is simply less restrictive, and also includes some additional options. Quick summary:

- Order is no longer important, so `/+ISC/2010-2014/15d1537/license.txt` === `/15d1537/+ISC/2010-2014/license.txt` === /license.txt/15d1537/+ISC/2010-2014` ... etc
- Addition of license type selection using `+LICENSE_TYPE` (supports only `+MIT` and `+ISC` atm)
- Ability to call dates with or without `@` prefix (`2010` === `@2010` and `2010-2012` === `@2010-2012`)
- Ability to leave out ending date and have it automatically use current year, so using `@2010-` (notice the trailing dash without a second date) would result in `2010 - 2016` (or whatever the current year is).
- I think that is it...

There are no *intentional*  BC breaks in URL handling.

**Comment by robfrawley** [comment]
The ephemeral disk is an interesting point... any ideas as to how to handle JSON POSTing moving forward? (Do you *want* to continue supporting it?)

**Comment by remy** [comment]
Re: JSON posting, I'd like to, but it's a nicety that I can afford to lose in the service for now. I think longer term it'll do something like do a github authenticated commit to this repo (but that can all be a separate branch in favour of keeping this branch focused).

**Comment by robfrawley** [comment]
Re-based. Removed JSON POST logic completely.

**Pull request closed without merging**